app/services/degrees/dfe_reference.rb

```diff
@@ -25,12 +25,25 @@ def find_grade(uuid: nil, name: nil, hesa_code: nil)
       end
 
       def find_item(list_type, filters)
-        DfE::ReferenceData::Degrees.const_get(list_type.to_s.upcase).all.find do |record|
+        refdataset = DfE::ReferenceData::Degrees.const_get(list_type.to_s.upcase)
+
+        refdataset.all.find do |record|
           filters.compact.any? do |field, value|
-            same_string?(record[field], value)
+            if field.to_s.include?("name") || field.to_s.include?("abbreviation")
```

Comment:
We probably don't need to check this as they're in they're in the match values. Should this method default to finding by id (or hecos|hesa code) by default as they should be unambiguous and then just fallback to the match values?

Reply:
@gpeng i think I've implemented this now if you want to take another look